app/main/views/direct_award.py

```diff
@@ -251,3 +304,31 @@ def lock_project(project_id):
     db.session.commit()
 
     return jsonify(project=project.serialize())
+
+
+@main.route('/direct-award/projects/<int:project_id>/record-download', methods=['POST'])
+def record_project_download(project_id):
+    updater_json = validate_and_return_updater_request()
+
+    project = DirectAwardProject.query.filter(DirectAwardProject.id == project_id).first_or_404()
+
+    # Up for discussion: should this endpoint abort if downloaded_at is already set, or should it continue (so that
+    # and audit event is created for every download)?
```

Comment:
FWIW I think if we are having a separate endpoint, then the expectation is that you can call it separately to getting the download? In which case it would really make sense to audit every download.

In terms of the language on the frontend, I think (but @karlchillmaid may differ) that if we are talking about "when was this thing downloaded", then the _assumption_ would be "when _last_ did that happen". And if you meant "when was the project _first_ downloaded" then we'd have needed to call that out.

But I'm not quite sure why we're recording this anyway, except so that we can have one extra step that can get ticked :-)

Reply:
That's really the only reason - so that the last state of the tasklist is unlocked. We don't ever show on the frontend the datetime of the download, so it doesn't matter overly much if we record the first or last timestamp. Happy to change if you'd prefer.

Reply:
I guess I'd marginally prefer to just keep recording the last date, yeah? But makes little odds if we aren't showing it for sure.

Reply:
I guess we'll have the audit events if we ever need to go back and see, so sure.